web/models/anagrams.ex

```diff
@@ -30,30 +28,17 @@ defmodule Anagrams do
     |> Enum.sort
   end
 
-  def dictionary(human_readable_dictionary) do
-    dictionary(human_readable_dictionary, %{})
-  end
-
-  defp dictionary([], output_dict) do
-    output_dict
-  end
-
   # returns map with entries like ["d", "g", "o"] => ["god", "dog"]
-  defp dictionary([current_word | remaining_words], output_dict) do
-    entry_letterbag = letterbag(current_word)
-    if Map.has_key?(output_dict, entry_letterbag) do
-      existing_entry = Map.get(output_dict, entry_letterbag)
-      updated_dict = Map.put(output_dict, entry_letterbag, existing_entry ++ [current_word])
-    else
-      updated_dict = Map.put(output_dict, entry_letterbag, [current_word])
-    end
-    dictionary(remaining_words, updated_dict)
+  def dictionary(human_readable_dictionary) do
+    Enum.reduce(human_readable_dictionary, %{}, fn word, map_acc ->
```

Comment:
Seriously, my brain is too tired to comprehend this right now but I'd like you to walk me through it.